packages/mini-browser/src/browser/location-mapper-service.ts

```diff
@@ -115,7 +115,7 @@ export class FileLocationMapper implements LocationMapper {
         return location.startsWith('file://') ? 1 : 0;
     }
 
-    map(location: string): MaybePromise<string> {
+    map(location: string): string {
```

Comment:
The idea behind the `MaybePromise` was to allow subclasses to do something else which might not be synchronous. Cannot you use the API in an async way?

Reply:
The hook is in the markdown renderer for images and it's all synchronous, cf. https://github.com/theia-ide/theia/blob/f5307d53c1a94392aa7b40c86f31fbb1d0707f53/packages/preview/src/browser/markdown/markdown-preview-handler.ts#L238-L248.

Alternatively re-parsing and patching the output after rendering sounds inefficient.